### PR TITLE
BEFOUND-282 Adopt ArchUnit rules from SSDK

### PR DESCRIPTION
**Pull request by lehphyro**
Depends on this SSDK PR: https://stash.backbase.com/projects/SSDK/repos/service-sdk/pull-requests/835/overview

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -10,7 +10,7 @@
     <parent>
         <artifactId>service-sdk-starter-core</artifactId>
         <groupId>com.backbase.buildingblocks</groupId>
-        <version>12.2.0</version>
+        <version>13.0.0-cr.13-SNAPSHOT</version>
```
Comment (danielabeledo) [inline pom.xml:13]: this is not found in repo yet, right?

File: services/review/src/test/resources/archunit_ignore_patterns.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Ignore configuration classes because multi-tenancy support is not done yet: https://github.com/Backbase/golden-sample-services/issues/6
+.*com\.backbase\.goldensample\.review\.config.*
+
+# Ignore service classes because security is not done yet: https://github.com/Backbase/golden-sample-services/issues/4
+.*com\.backbase\.goldensample\.review\.service.*
+
+# Ignore controllers due to the way multi-version support was implemented
```
Comment (lehphyro) [inline services/review/src/test/resources/archunit_ignore_patterns.txt:7]: I think we should introduce a domain layer to separate API representations from persistence ones in order to support multiple API versions. The related violations are being ignored for now.

**Review by pa314159 — APPROVED** [review]

File: services/product/pom.xml [hunk 3]
```diff
@@ -129,6 +129,24 @@
             <artifactId>mariadb-java-client</artifactId>
             <scope>test</scope>
         </dependency>
+
+        <!-- Start - ArchUnit rule checks -->
+        <dependency>
+            <groupId>com.backbase.buildingblocks</groupId>
+            <artifactId>service-sdk-archunit</artifactId>
```
[the remark this answers is not in the document]
Comment (lehphyro) [reply services/product/pom.xml:136]: Fixed.

File: services/review/src/test/resources/archunit_ignore_patterns.txt (hunk 2 above)
Comment (lehphyro) [reply services/review/src/test/resources/archunit_ignore_patterns.txt:7]: I can do that after this PR is merged.

**Comment by lehphyro** [comment]
I don't know why `com.backbase.goldensample.store.configuration.AppConfigurationMultitenancyTest` is failing, maybe @christopherm-backbase can help fix the multi-tenancy issue?

**Comment by christopherm-backbase** [comment]
Updated to fix some failing tests and SSDK 13.0.0 is now on repo.

**Comment by lehphyro** [comment]
> Updated to fix some failing tests and SSDK 13.0.0 is now on repo.

Thank you. Shall we merge this now?

**Review by christopherm-backbase — APPROVED** [review]
Happy to be merged.

**Pull request merged**